daml/Marketplace/Clearing/Service.daml

```diff
@@ -0,0 +1,212 @@
+{-# LANGUAGE MultiWayIf #-}
+
+module Marketplace.Clearing.Service where
+
+import DA.Functor (void)
+
+import DA.Finance.Asset
+import DA.Finance.Types
+
+import qualified Marketplace.Rule.AllocationAccount as AllocationAccountRule (T, Deposit(..), Withdraw(..))
+import Marketplace.Clearing.Model
+import Marketplace.Clearing.Utils
+
+template Service
+  with
+    operator : Party
+    provider : Party
+    customer : Party
+    ccpAccount : Account
+    clearingAccount : Account
+    marginAccount : Account
+  where
+    signatory operator, provider, customer
+
+    key (operator, provider, customer) : (Party, Party, Party)
+    maintainer key._1
+
+    let
+      transferToMarginAccount : ContractId AssetDeposit -> Update (ContractId AssetDeposit)
+      transferToMarginAccount = \depositCid -> exerciseByKey @AllocationAccountRule.T marginAccount.id
+          AllocationAccountRule.Deposit with ..
```

Comment:
Can you give me some further clarification on this? I've used defining the explicit variable instead of `..` in another model that I've built as I find it more readable, but I see `exercise X.Choice with ..` in 95% of the rest of the daml in this project so I went with `..`.

I'm happy to change it, but I'm curious why it may be an issue here instead of other places, as the only variable here is `depositCid` (which is defined right above) so it seems clear to me what the `..` means in this context.

Reply:
I think just use your personal judgement. In this case if there is a single variable, I think it more readable just to use the name.
Also the thing to watch out for is shadowing, i.e. variable re-assignment between `..`'s in a `do` loop for example, as that makes it really confusing.

Reply:
> as the only variable here is depositCid (which is defined right above) 

I see now what you mean; I confused the successive functions (which all use `..`) as being part of the same block.

Btw I think you can shorten it to something like this:

```
     transferToMarginAccount : ContractId AssetDeposit -> Update (ContractId AssetDeposit)
     transferToMarginAccount = exerciseByKey @AllocationAccountRule.T marginAccount.id . AllocationAccountRule.Deposit
```

Reply:
Nice! I didn't realize that choices were actually just constructor functions that are composable - the `with` sugar threw me off, so that's pretty cool. 

I also tend to avoid shadowing in general if I can help it, we had `-Wall` on at my last job :).